src/WebJobs.Extensions.Http/ClaimsIdentityBindingProvider.cs

```diff
@@ -0,0 +1,167 @@
+        }
+
+        private class ClaimsIdentityBinding : IBinding
+        {
+            private readonly ParameterInfo _parameter;
+
+            public ClaimsIdentityBinding(ParameterInfo parameter)
+            {
+                _parameter = parameter;
+            }
+
+            public bool FromAttribute
+            {
+                get { return false; }
+            }
+
+            public async Task<IValueProvider> BindAsync(BindingContext context)
+            {
+                if (context == null)
+                {
+                    throw new ArgumentNullException("context");
+                }
+
+                // if the request is EasyAuth authenticated, get the identity provider type
+                object value = null;
+                string identityProvider = null;
+                if (context.BindingData.TryGetValue(HttpTriggerAttributeBindingProvider.HttpHeadersKey, out value))
+                {
+                    IDictionary<string, string> headers = (IDictionary<string, string>)value;
+                    if (headers != null)
+                    {
+                        headers.TryGetValue(HttpConstants.AntaresEasyAuthProviderHeaderName, out identityProvider);
+                    }  
+                }
+
+                var identity = GetPrimaryIdentity(ClaimsPrincipal.Current, identityProvider);
+                return await BindInternalAsync(identity);
+            }
+
+            public Task<IValueProvider> BindAsync(object value, ValueBindingContext context)
+            {
+                if (context == null)
+                {
+                    throw new ArgumentNullException("context");
+                }
+
+                var identity = GetPrimaryIdentity(ClaimsPrincipal.Current);
+
+                return BindInternalAsync(identity);
+            }
+
+            private static Task<IValueProvider> BindInternalAsync(ClaimsIdentity identity)
+            {
+                return Task.FromResult<IValueProvider>(new ClaimsIdentityValueProvider(identity));
+            }
+
+            public ParameterDescriptor ToParameterDescriptor()
+            {
+                return new ParameterDescriptor
+                {
+                    Name = _parameter.Name,
+                    DisplayHints = new ParameterDisplayHints
+                    {
+                        Description = "ClaimsIdentity"
+                    }
+                };
+            }
+
+            private static ClaimsIdentity GetPrimaryIdentity(ClaimsPrincipal claimsPrincipal, string identityProvider = null)
+            {
+                // if the principal contains a key based identity that will take precedence
+                // over any other identity
+                var identity = claimsPrincipal.Identities.LastOrDefault(p => p.IsAuthenticated && string.Compare(p.AuthenticationType, "key", StringComparison.OrdinalIgnoreCase) == 0);
```

Comment:
Why `string.Compare` and not `string.Equals` (here and in the logic further below)?